presto-druid/src/main/java/com/facebook/presto/druid/DruidSplitManager.java

```diff
@@ -21,21 +28,31 @@
 import com.facebook.presto.spi.connector.ConnectorSplitManager;
 import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
 import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
 
 import javax.inject.Inject;
 
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 import static com.facebook.presto.druid.DruidSessionProperties.isComputePushdownEnabled;
 import static com.facebook.presto.druid.DruidSplit.createBrokerSplit;
 import static com.facebook.presto.druid.DruidSplit.createSegmentSplit;
 import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 public class DruidSplitManager
         implements ConnectorSplitManager
 {
     private final DruidClient druidClient;
+    // Druid stores segments times as strings in ISO-8601 format in UTC down to fractional seconds.
+    private DateTimeFormatter timeSegmentFormatter = new DateTimeFormatterBuilder().appendInstant(3).parseCaseInsensitive().toFormatter();
```

Comment:
I think it might be easier to use org.joda.time.format.ISODateTimeFormat rather than build our own iso time formatter.

Reply:
I'm not familiar with joda but I'll take a look at it. I tried using a stock time formatter from java utils, but they didn't have one in the exact same pattern as how druid stores it. I'll see if joda has one like that. If not then I'll convert this to be a static one.